packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,254 @@
+
+![ ](spec-images/select-alternative-combobox.png)
+
+> NOTE: SEE FUTURE CONSIDERATIONS FOR MULTI-SELECTION
+
+To allow end-users to multi-select menu items, use a [**tag picker**](/packages/nimble-components/src/tag-picker/specs/IxD.md).
+
+![ ](spec-images/select-alternative-picker.png)
+
+## Design
+### Configuration
+#### Label
+> NOTE: NEEDS VISUAL DESIGN FOR PLACEMENT
+
+The client-user should be able to specify whether they want the label to be positioned at the top or to the left of the select or not at all. Selects that don't have a label should still have an ARIA label.
+
+![ ](spec-images/select-6.png)
+
+#### Icons
+The client-user should be able to specify whether they want left icons on menu items or not. Icons show up in menu items and are displayed as part of the value in the field button. Images and other content not specified here should not be permitted.
+
+![ ](spec-images/select-7.png)
+
+#### Groups
+> QUESTION: IS THE SL "VIEW" PICKER A SELECT (SHOULD IT BE)? WHAT ABOUT GROUPS WITHOUT LABELS, WHAT ABOUT A SELECTION GROUP VS. AN ACTION GROUP OF MENU ITEMS (A.K.A MEGA-MENUS)?
+
+The client-user should be able to create groups for menu items in the menu. Groups are not interactable.
+
+![ ](spec-images/select-8.png)
+
+#### Secondary Text
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify whether they want secondary text on menu items or not.
+
+![ ](spec-images/select-9.png)
+
+The client-user should be able to specify whether they want other information (including groups) displayed as secondary text in the field input.
+
+![ ](spec-images/select-10.png)
+
+#### Selection
+> NOTE: NOT SURE THIS IS A REAL USE CASE FOR SELECT COMPONENT, SEE FUTURE CONSIDERATIONS FOR MULTI-SELECTION
+
+The client-user should be able to specify whether they want to display selection with or without a check mark.
+
+![ ](spec-images/select-11.png)
+
+#### Filter
+The client-user should be able to specify whether they want filtering available in the select component. The filter field appears in the menu.
+
+![ ](spec-images/select-12.png)
+
+### Behavior
+#### Minimum Width
+> NOTE: NEEDS DEV REVIEW
+
+The field button minimum width is three times the height of the component (to make components targetable on touch screens). The menu width is the same as the field button until reaching the minimum width (without truncating) of the longest string in the menu items.
+
+![ ](spec-images/select-13.png)
+
+#### Menu Height
+The menu height should be enough to display five to seven menu items before overflowing with a vertical scrollbar (not pictured).
+
+#### Overflow Text
+> NOTE: NEEDS DEV & VISUAL DESIGN REVIEW
+
+All text in the select component should truncate, not wrap.
+
+![ ](spec-images/select-14.png)
+
+#### Progress & Loading
+When displaying menu items in the menu lags or is expected to take longer than normal, the menu should indicate that it's loading. The "Loading…" display is not interactive.
```

Comment:
I'm not convinced that this is a concern that the `Select` component can adequately manage on its own (particularly when we consider the below feature of a "Load more" button). My preference would be that, if anything, the `Select/Combobox` offer a slot that allows users to populate the bottom of the dropdown with user-defined content.

My current conception of what the APIs might look like if we were to support this in the `Select/Combobox` directly would allow it to be configured to be in an awkward state, relying on the user to tightly manage so that it displays things properly, which I would rather avoid. While the slot API that allows general content also enables it to be configured awkwardly, the API itself isn't explicit about it, and is instead "explicit" in its expectation that it's up to the user to configure it appropriately.